### PR TITLE
Refactor Lahman and fix separator bug in windows

### PR DESCRIPTION
**Pull request by TheCleric**
There is presently a bug in the lahman library for Windows users with no downloaded cache of the data. This is due to the fact that we use os.sep to build the file paths, but internal to the zip file, it is always `/` regardless of os.

In addition, I've refactored this code to remove a lot of the common code among the functions.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -145,3 +145,6 @@ cython_debug/
 
 # MacOS
 .DS_Store
+
+# lahman cache
+baseballdatabank-master
```
Comment (schorrm) [inline .gitignore:150]: Whoa whoa whoa

Comment (TheCleric) [reply .gitignore:150]: Is this not preferred? I'm just preventing an accidental checkin of anyone downloading lahman while working on the library. If we want to leave this to enforcement at the PR level we certainly can. Just trying to add some security.

**Comment by schorrm** [comment]
I think there's a much cleaner way to go about this, mind if I take a shot?

File: .gitignore (hunk 1 above)
Comment (schorrm) [reply .gitignore:150]: fair enough, but this **really** shouldn't happen

**Comment by schorrm** [comment]
I think it made a bunch of things overly complex

**Comment by schorrm** [comment]
Closing, #118

**Pull request closed without merging**